syntax = "proto3";

package football_commentary;

// Service for the Commentary service to send text to the TTS service
service CommentaryToTts {
  rpc SynthesizeSpeech (TextToSpeechRequest) returns (SpeechSynthesisResponse);
}

message TextToSpeechRequest {
  string text = 1;
  string language_code = 2; // e.g., "en-US", "es-ES"
  string voice_name = 3;    // Optional: specific voice if supported by TTS
  string request_id = 4;    // For tracking
}

message SpeechSynthesisResponse {
  string status = 1;        // e.g., "PROCESSING", "COMPLETED", "ERROR"
  bytes audio_content = 2; // The synthesized audio bytes (if returned directly)
                          // Alternatively, this could be a URI or ID to fetch the audio
  string audio_format = 3;  // e.g., "wav", "mp3", "pcm"
  string request_id = 4;    // Echoing the request_id
} 